django/db/models/base.py

```diff
@@ -1087,7 +1087,8 @@ def _perform_unique_checks(self, unique_checks):
             for field_name in unique_check:
                 f = self._meta.get_field(field_name)
                 lookup_value = getattr(self, f.attname)
-                if lookup_value is None:
+                if (lookup_value is None or
+                        lookup_value == '' and connection.features.interprets_empty_strings_as_nulls):
```

Comment:
As the checks are performed against `model_class._default_manager` the `connections[router.db_for_read(model_class)]` connection should be used.